src/main/java/parsing/Parser.java

```diff
@@ -118,8 +153,10 @@ public NodeGraph parse(final File file, NodeGraph graph) {
                                 out.flush();
                                 line2 = line2.substring(line2.indexOf('\t') + 1);
                                 line2 = line2.substring(line2.indexOf('\t') + 1);
-                                String nodeGenomes = line2.substring(0, line2.indexOf('\t'));
-                                addGenomes(gw, nodeGenomes);
```

Comment:
No, The fist invocation is a special one, that one writes all genomes in the gfa file, while that other in the while loop write for every node the indexes of the genomes that visit these particular nodes.
Thus these two are different and should both stay.